File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        a = abs(x)
        rev = 0
        INT_MAX = 2**31 - 1
        while a > 0 :
            rev = rev*10 + a%10
            a //= 10
        if rev > INT_MAX : 
            return 0
        if x < 0 :
            return -1 * rev
        return rev

File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        int a = Math.abs(x);

        int rev = 0;

        while(a > 0) {
            int digit =  a%10;
            a /= 10;
            if(rev  > (Integer.MAX_VALUE - digit) / 10) {
                return 0;
            }
            rev = rev*10 + digit;
        }

        return x < 0 ? -rev : rev;
    }
}